### ISSUE TITLE
Feature request: operator "sparse softmax"

### ISSUE THREAD

**Issue by eric-haibin-lin** [apache/incubator-mxnet#12729]
The sparse softmax is useful for the transformer model: https://www.tensorflow.org/api_docs/python/tf/sparse_softmax

**Comment by eric-haibin-lin** [comment]
Can be useful for many attention based models.
